packages/shared/src/utils/validation/declarative-validation-rules.ts

```diff
@@ -62,17 +62,17 @@ export const validNumber = (value: string, opts: TOptions) => {
         message = form_error_messages.number();
     } else if ('min' in options && 'max' in options && +options.min === +options.max && +value !== +options.min) {
         is_ok = false;
-        message = form_error_messages.value(addComma(options.min));
+        message = form_error_messages.value(addComma(options.min, options.decimals));
     } else if (
         'min' in options &&
         'max' in options &&
         options.min > 0 &&
         (+value < +options.min || isMoreThanMax(+value, options))
     ) {
         is_ok = false;
-        const min_value = addComma(options.min);
-        const max_value = addComma(options.max);
-        message = form_error_messages.betweenMinMax(min_value, max_value, options.decimals);
+        const min_value = addComma(options.min, options.decimals);
+        const max_value = addComma(options.max, options.decimals);
+        message = form_error_messages.betweenMinMax(min_value, max_value);
```

Comment:
`addComma` function can take **decimals** parameter to proper format the value that we can show on UI. It also works fine for numbers with scientific notation.